onedal/_device_offload.py

```diff
@@ -166,21 +203,11 @@ def _run_on_device(func, obj=None, *args, **kwargs):
     return func(*args, **kwargs)
 
 
-if dpnp_available:
-
-    def _convert_to_dpnp(array):
-        if isinstance(array, usm_ndarray):
-            return dpnp.array(array, copy=False)
-        elif isinstance(array, Iterable):
-            for i in range(len(array)):
-                array[i] = _convert_to_dpnp(array[i])
-        return array
-
-
-def support_usm_ndarray(freefunc=False, queue_param=True):
+def support_array_api(freefunc=False, queue_param=True):
```

Comment:
Does this handle dpnp? If so then the name is a misnomer (since it doesn't have array_api support).

Reply:
Currently, yes. But it is in DPNP plans to cover this some how. We also have some development for DPNP case, so I think make sense to leave it as is.

Reply:
Though at the moment it is misleading, and dependent on a promise without a promised delivery date. Please rename.

Reply:
dpnp doesn't yet follow array API, but dpctl, that is the dependent of the dpnp is following.
Currently, we are not using dpnp's dpctl array api for the case.
I already have a ticket for dpnp array api enabling in case, that will be used only on sklearnex side. so in this case I think make sense to leave it as is.

Reply:
We can change the name if it is required.
Possible name is `support_input_format`. It could be common also for Dataframe API functionally handling.
What is your thoughts?

Reply:
Already renamed, please check.